File: 连续子数组的最大和.py
# -*- coding:utf-8 -*-
class Solution:
    def __init__(self):
        self.re=[]
    def FindGreatestSumOfSubArray(self, array):
        # write code here
        a=[]
        if len(array)==1:
            self.re.append(array[0])
            return 
        count=0
        for each in array:
            count+=each
            a.append(count)
        self.re.extend(a)
        self.FindGreatestSumOfSubArray(array[1:])
        return max(self.re)

File: 剑指offer（c++）/连续子数组的最大和.cpp
class Solution {
public:
    int FindGreatestSumOfSubArray(vector<int> array) {
        int count=0;
        int Max=-9999999;
        for(int i = 0 ;i<array.size();i++){
            if(count<=0){
                count=array[i];
            }
            else{
                count+=array[i];
            }
            if(Max<count)
                Max=count;
        }
        return Max;
    }
};